doc/tutorials/trustyACRN.rst

```diff
@@ -0,0 +1,408 @@
+- Trusted applications or services that run on the TEE/Trusty OS in
+  secure world;
+- Untrusted applications running in normal world that use services
+  provided by Trusted applications.
+
+Software running in normal world can use Trusty client library APIs to
+connect to trusted applications and exchange arbitrary messages with
+them, just like a network service over IP. It is up to the application
+to determine the data format and semantics of these messages using an
+app-level protocol. Reliable delivery of messages is guaranteed by the
+underlying Trusty infrastructure (Trusty Drivers), and the communication
+is completely asynchronous.
+
+Although this Trusty infrastructure is built by Google for Android OS,
+it can be applied to any normal world OS (typically a Linux-based OS).
+The Trusty OS infrastructure in secure world is normal world
+OS-agnostic. The differences truly depend on the security services that
+normal world OS would like to have.
+
+Trusty Services
+***************
+
+There are many uses for a Trusted Execution Environment such as mobile
+payments, secure banking, full-disk encryption or file-based encryption,
+multi-factor authentication, device reset protection, replay-protected
+persistent storage (secure storage), wireless display ("cast") of
+protected content, secure PIN and fingerprint processing, and even
+malware detection.
+
+In embedded products such as an automotive IVI system, the most important
+security services requested by customers are keystore and secure
+storage. In this article we will focus on these two services.
+
+Keystore
+========
+
+Keystore (or Keymaster app in Trusty OS) provides the following
+services:
+
+- Key generation
+- Import and export of asymmetric keys (no key wrapping)
+- Import of raw symmetric keys (no key wrapping)
+- Asymmetric encryption and decryption with appropriate padding modes
+- Asymmetric signing and verification with digesting and appropriate
+  padding modes
+- Symmetric encryption and decryption in appropriate modes, including
+  an AEAD mode
+- Generation and verification of symmetric message authentication codes
+
+Protocol elements, such as purpose, mode and padding, as well as access
+control constraints, are specified when keys are generated or imported
+and are permanently bound to the key, ensuring the key cannot be used in
+any other way.
+
+In addition to the list above, there is one more service that Keymaster
+implementations provide, but which is not exposed as an API: Random
+number generation. This is used internally for generation of keys,
+Initialization Vectors (IVs), random padding, and other elements of
+secure protocols that require randomness.
+
+Using Android as an example, Keystore functions are explained in greater
+details in this `Android keymaster functions document
+<https://source.android.com/security/keystore/implementer-ref>`_
+
+.. figure:: images/trustyacrn-image3.png
+   :align: center
+   :width: 600px
+   :name: keymaster-app
+
```

Comment:
Could we also have a title for this figure like trusty-arch? We can name this "Keymaster Functions"

Reply:
Agree. seems Figure XXX is missing if no title.

Reply:
name could be "Keystore service" or just simply "keystore"